calico.go

```diff
@@ -319,24 +319,39 @@ func cmdDel(args *skel.CmdArgs) error {
 		return err
 	}
 
-	if err := calicoClient.WorkloadEndpoints().Delete(api.WorkloadEndpointMetadata{
+	ep := api.WorkloadEndpointMetadata{
 		Name:         args.IfName,
 		Node:         hostname,
 		Orchestrator: orchestrator,
-		Workload:     workload}); err != nil {
+		Workload:     workload}
+	err = calicoClient.WorkloadEndpoints().Delete(ep)
+	if _, ok := err.(errors.ErrorResourceDoesNotExist); ok {
+		logger.WithField("endpoint", ep).Warn("Asked to delete endpoint object, but it doesn't exist. Ignoring.")
+	} else {
 		return err
 	}
 
 	// Only try to delete the device if a namespace was passed in.
 	if args.Netns != "" {
-		fmt.Fprintf(os.Stderr, "Calico CNI deleting device in netns %s\n", args.Netns)
-		err = ns.WithNetNSPath(args.Netns, func(_ ns.NetNS) error {
-			_, err = ip.DelLinkByNameAddr(args.IfName, netlink.FAMILY_V4)
+		// Ensure the namespace & device exist.
+		nilWhenDevExists := ns.WithNetNSPath(args.Netns, func(_ ns.NetNS) error {
```

Comment:
@tomdee We have to manually check if the interface exists here because CNI library doesn't return different error types depending on the error encountered during [`DelLinkByNameAddr`](https://github.com/containernetworking/cni/blob/master/pkg/ip/link.go#L157-L173). If that function did a `ErrDeviceNotFound`, `ErrNamespaceNotFound`, `ErrCouldntDeleteDevice`, we could gracefully decide if we needed to error or not.